Sources/SwiftProtobuf/Message+FieldMask.swift

```diff
@@ -0,0 +1,126 @@
+// Sources/SwiftProtobuf/Message+FieldMask.swift - Message field mask extensions
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extend the Message types with FieldMask utilities.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+
+  /// Checks whether the given path is valid for Message type.
+  ///
+  /// - Parameter path: Path to be checked
+  /// - Returns: Boolean determines path is valid.
+  public static func isPathValid(
+    _ path: String
+  ) -> Bool {
+    var message = Self()
+    return message.hasPath(path: path)
+  }
+
+  internal mutating func isPathValid(
+    _ path: String
+  ) -> Bool {
+    hasPath(path: path)
+  }
+}
+
+/// Defines available options for merging two messages.
+public struct MergeOption {
```

Comment:
This ends up a top level type in the module, I'm wondering if it should be scoped or named in such a way as to scope it.  It seems like "MergeOptions" is pretty generic and someone might expect it apple to other apis?

The two options that jump to mind are either name it with _FieldMask_ in some way, or declare it scoped to the FieldMask message.

@tbkka any thoughts on better approaches?

Reply:
My initial suggestion would just be:

```swift

extension Google_Protobuf_FieldMask {
  public struct MergeOptions {
    // ...
  }
}
```

That way it is (hopefully) clear that it is specific to FieldMask.

Reply:
Done. Also other comments are resolved. could you please check all again?